zhaquirks/tuya/ts0601_din_power.py

```diff
@@ -177,6 +183,78 @@ def _update_attribute(self, attrid, value):
             self.endpoint.electrical_measurement.power_factor_reported(value / 10)
 
 
+class ZemismartManufCluster(TuyaManufClusterAttributes):
+    """Manufacturer Specific Cluster of the Zemismart SPM01 Power Meter device."""
+
+    attributes = {
+        ZEMISMART_TOTAL_ENERGY_ATTR: ("energy", t.uint32_t, True),
+        ZEMISMART_TOTAL_REVERSE_ENERGY_ATTR: ("reverse_energy", t.uint32_t, True),
+        ZEMISMART_VCP_ATTR: ("vcp_raw", t.data64, True),
+    }
+
+    def _update_attribute(self, attrid, value):
+        super()._update_attribute(attrid, value)
+        if attrid == ZEMISMART_TOTAL_ENERGY_ATTR:
+            self.endpoint.smartenergy_metering.energy_deliver_reported(value)
+        elif attrid == ZEMISMART_TOTAL_REVERSE_ENERGY_ATTR:
+            self.endpoint.smartenergy_metering.energy_receive_reported(value)
+        elif attrid == ZEMISMART_VCP_ATTR:
+            self.endpoint.electrical_measurement.voltage_reported(
+                (value[7] * 256) + value[6]
+            )
+            self.endpoint.electrical_measurement.current_reported(
+                (value[5] * 256 * 256) + (value[4] * 256) + value[3]
+            )
+            self.endpoint.electrical_measurement.power_reported(
+                (value[2] * 256 * 256) + (value[1] * 256) + value[0]
+            )
```

Comment:
There's a comment further below about this, but we can call `update_attribute` on another cluster (from another cluster) now.

So something like this should work for example:
```python
self.endpoint.electrical_measurement.update_attribute(ElectricalMeasurement.AttributeDefs.active_power.id, (value[5] * 256 * 256) + (value[4] * 256) + value[3])
```